src/site/_data/patterns.js

```diff
@@ -0,0 +1,125 @@
+/* global __basedir */
+
+const nunjucks = require('nunjucks');
+const fs = require('fs');
+const path = require('path');
+
+module.exports = {
+  // Grabs all patters that it can find at the root level then builds up a dataset,
+  // rendered markup, view markup and docs. Lastly, it finds any variants and makes
+  // those part of the pattern, too
+  get items() {
+    const basePath = path.resolve(__basedir, 'src/pattern-library/patterns');
+
+    // Grabs each folder inside patterns, excluding hidden files/folders
+    const patterns = fs
+      .readdirSync(basePath)
+      .filter((item) => !/(^|\/)\.[^/.]/g.test(item));
+
+    // For creating a result collection
+    const result = [];
+
+    // This is used for both patterns and variants to grab markup, data and docs
+    const buildPattern = (patternPath, patternName) => {
+      const response = {};
+
+      if (!fs.existsSync(path.resolve(patternPath, `${patternName}.njk`))) {
+        return null;
+      }
+
+      response['markup'] = fs.readFileSync(
+        path.resolve(patternPath, `${patternName}.njk`),
+        'utf8',
+      );
```

Comment:
This is a good catch mate. I think we should log something out to the console for sure, but also, we should check the response where this function is called because if `!buildPattern()`, the pattern shouldn't be added to the end response

Reply:
Second thought: this will never be hit because the original lookup for patterns is the `.njk` files themselves, so I'll just delete the check.